### DISCUSSION TITLE
Redirect back to initial route after signin

### DISCUSSION THREAD

**Discussion by martijnhalekor** [manchenkoff/nuxt-auth-sanctum#255] [Q&A] [answered]
Hey! I was wondering if it's possible to configure the following behaviour:

1. Visit `example.com/specific-route`
2. Navigate to `example.com/login` by clicking on a login button in the UI
3. Submit credentials.
4. Get redirected to `example.com/specific-route`

Currently, I always end up at `example.com/` because my redirect.onLogin is configured to `'/'`. Can this also be a dynamic value?

It feels like I'm looking for something this package alone can't provide, but any tips or examples are greatly appreciated 🙏

**Comment by manchenkoff** [comment, marked as answer]
Hey @martijnhalekor, yes, it is possible if `specific-route` requires authentication. You can set [`sanctum.redirect.keepRequestedRoute`](https://manchenkoff.gitbook.io/nuxt-auth-sanctum/usage/configuration) to `true` in your `nuxt.config.ts` and it will keep this URL in query params for redirect back to it once credentials are sent.

In other cases, it is better to set the `onLogin` route to `false` and handle it in your code if you want to use a dynamic value.
